components/automate-cli/cmd/chef-automate/status.go

```diff
@@ -234,7 +280,267 @@ func runStatusCmd(cmd *cobra.Command, args []string) error {
 	}
 }
 
+type statusCmdFromBastionHelper interface {
+	getAutomateHAInfraDetails() (*AutomateHAInfraDetails, error)
+	isManagedServicesOn() bool
+	executeRemoteExecutor(*NodeTypeAndCmd, SSHUtil, *cli.Writer) (map[string][]*CmdResult, error)
+	printStatusOutput(map[string][]*CmdResult, string, *sync.WaitGroup, *sync.Mutex, *cli.Writer)
+}
+
+type statusCmdFromBastionHelperImpl struct{}
+
+func NewStatusCmdFromBastionHelperImpl() *statusCmdFromBastionHelperImpl {
+	return &statusCmdFromBastionHelperImpl{}
+}
+
+func (st *statusCmdFromBastionHelperImpl) getAutomateHAInfraDetails() (*AutomateHAInfraDetails, error) {
+	return getAutomateHAInfraDetails()
+}
+
+func (st *statusCmdFromBastionHelperImpl) isManagedServicesOn() bool {
+	return isManagedServicesOn()
+}
+
+func (st *statusCmdFromBastionHelperImpl) executeRemoteExecutor(nodemap *NodeTypeAndCmd, sshUtil SSHUtil, writer *cli.Writer) (map[string][]*CmdResult, error) {
+	remoteExecutor := NewRemoteCmdExecutor(nodemap, sshUtil, writer)
+	cmdResult, err := remoteExecutor.Execute()
+	return cmdResult, err
+}
+
+func (st *statusCmdFromBastionHelperImpl) printStatusOutput(cmdResult map[string][]*CmdResult, remoteService string, wg *sync.WaitGroup, mutex *sync.Mutex, writer *cli.Writer) {
+	mutex.Lock()
+	writer.Printf("\n=====================================================%s=======================================================\n", "Status on "+remoteService)
+	for _, value := range cmdResult {
+		for _, cmdResult := range value {
+			printOutput(remoteService, *cmdResult, []string{}, writer)
+		}
+	}
+	mutex.Unlock()
+	wg.Done()
+}
+
+type MockStatusCmdFromBastionHelperImpl struct {
```

Comment:
Shouldn't the mocking struct be present in the test file rather then a implementation file

Reply:
Done shifted in _test file